config/prow-staging/cluster/prowjob_customresourcedefinition.yaml

```diff
@@ -12,89 +12,96 @@
 # limitations under the License.
 
-apiVersion: apiextensions.k8s.io/v1beta1
+apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   name: prowjobs.prow.k8s.io
+  annotations:
+    "api-approved.kubernetes.io": "https://github.com/kubernetes/test-infra/pull/8669"
 spec:
   group: prow.k8s.io
-  version: v1
   names:
     kind: ProwJob
     singular: prowjob
     plural: prowjobs
   scope: Namespaced
-  validation:
-    openAPIV3Schema:
-      properties:
-        spec:
-          properties:
-            max_concurrency:
-              type: integer
-              minimum: 0
-            type:
-              type: string
-              enum:
-              - "presubmit"
-              - "postsubmit"
-              - "periodic"
-              - "batch"
-        status:
-          properties:
-            state:
-              type: string
-              enum:
-              - "triggered"
-              - "pending"
-              - "success"
-              - "failure"
-              - "aborted"
-              - "error"
-          anyOf:
-          - not:
-              properties:
-                state:
-                  type: string
-                  enum:
-                  - "success"
-                  - "failure"
-                  - "error"
-          - required:
-            - completionTime
-  additionalPrinterColumns:
-  - name: Job
-    type: string
-    description: The name of the job being run.
-    JSONPath: .spec.job
-  - name: BuildId
-    type: string
-    description: The ID of the job being run.
-    JSONPath: .status.build_id
-  - name: Type
-    type: string
-    description: The type of job being run.
-    JSONPath: .spec.type
-  - name: Org
-    type: string
-    description: The org for which the job is running.
-    JSONPath: .spec.refs.org
-  - name: Repo
-    type: string
-    description: The repo for which the job is running.
-    JSONPath: .spec.refs.repo
-  - name: Pulls
-    type: string
-    description: The pulls for which the job is running.
-    JSONPath: ".spec.refs.pulls[*].number"
-  - name: StartTime
-    type: date
-    description: When the job started running.
-    JSONPath: .status.startTime
-  - name: CompletionTime
-    type: date
-    description: When the job finished running.
-    JSONPath: .status.completionTime
-  - name: State
-    description: The state of the job.
-    type: string
-    JSONPath: .status.state
+  versions:
+  - name: v1
+    served: true
+    storage: true
+    schema:
+      openAPIV3Schema:
+        type: object
+        properties:
+          spec:
+            type: object
+            properties:
+              max_concurrency:
+                type: integer
+                minimum: 0
+              type:
+                type: string
+                enum:
+                - "presubmit"
+                - "postsubmit"
+                - "periodic"
+                - "batch"
+          status:
+            type: object
```

Comment:
```suggestion
            type: object
            x-kubernetes-preserve-unknown-fields: true
```

Reply:
From my local testing I noticed that `prowjob` CR created with PR only has `state` field under `status`, where a lot more fields are expected such as `startTime` etc. which very likely is the reason why the tests failed. Adding `x-kubernetes-preserve-unknown-fields: true` passed sinker tests at least. 

We will need to be more careful though on this change as this might need to be set elsewhere too.
@cjwagner  and @alvaroaleman  for awareness

Reply:
Oh ofc :facepalm: We also need the same for spec, what we have there currently is definitely not exhaustive.

I also tried to generate a complete schema for this at some point, but hit some roadblocks and didn't pursue it further: https://github.com/kubernetes/test-infra/pull/19522